robyn/__init__.py

```diff
@@ -456,9 +456,20 @@ def options(self, endpoint: str):
         return super().options(self.__add_prefix(endpoint))
 
 
-def ALLOW_CORS(app: Robyn, origins: List[str]):
+# keep the func params compatible with current one
+# use the type definition for compatible with python before 3.10
+def ALLOW_CORS(app: Robyn, origins: Union[List[str], str]):
     """Allows CORS for the given origins for the entire router."""
-    for origin in origins:
+
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMultipleAllowOriginNotAllowed
+    if isinstance(origins, list):
+        origin = origins[0]
```

Comment:
While this may avoid the issue, we still want all the origins to be considered.

Maybe we will have to include this test in the middleware implemented below.